homeassistant/components/mqtt_statestream/__init__.py

```diff
@@ -55,18 +39,11 @@
 
 async def async_setup(hass, config):
     """Set up the MQTT state feed."""
-    conf = config.get(DOMAIN, {})
+    publish_filter = config.get(DOMAIN)
+    conf = publish_filter.config
```

Comment:
I don't think we should use `vol.All` in the schema as as the domain level schema moves to `publish_filter.config` which isn't standard and will likely trip up other maintainers.

Its better to convert it here instead.

Reply:
Ok yea that's fair. It works when people include the filters as a separate field but when the filtering options are directly on the domain I can see how that causes confusion. Will make the adjustments.